traverse does not take an IterableOnce of Futures

philipschwarz opened the pull request:
Hello, I just want to draw your attention to the fact that the current description of the 'in' parameter seems incorrect. The most obvious  symptom of this is the fact that the description is identical to that of the 'in' parameter of the sequence method, but while the 'in' parameter of sequence is an `IterableOnce` of Futures, the 'in' parameter of traverse is not an `IterableOnce` of Futures, rather, its elements will be mapped to Futures by invoking the function that is the 'fn' parameter. I am not suggesting that my changes be made verbatim, you'll certainly do a much better job of figuring out an 'appropriate' way of correcting the description (e.g. one consistent with the rest of the Scaladoc).

I appreciate that it is not easy to write this kind of description.

Thanks.

philipschwarz committed:
traverse does not take an IterableOnce of Futures

Hello, I just want to draw your attention to the fact that the current description of the 'in' parameter seems incorrect. The most obvious  symptom of this is the fact that the description is identical to that of the 'in' parameter of the sequence method, but while the 'in' parameter of sequence is an `IterableOnce` of Futures, the 'in' parameter of traverse is not an `IterableOnce` of Futures, rather, its elements will be mapped to Futures by invoking the function that is the 'fn' parameter. I am not suggesting that my changes be made verbatim, you'll certainly do a much better job of figuring out an 'appropriate' way of correcting the description (e.g. one consistent with the rest of the Scaladoc).

viktorklang commented:
Thanks for your PR!

Technically, since it returns a subtype of iterableOnce, I'm convince we can improve it even further.
Do you have a suggestion which would cover that part as well?

lrytz commented:
Can we just say "collection"? That would be easy enough.

philipschwarz commented:
@viktorklang no, sorry, I don't have a suggestion and I don't know what you are envisaging - can you please elaborate?

philipschwarz commented:
@lrytz Hello Lukas - should I take your words literally, i.e. are you suggesting the following?
```Scala
   * @param in  the collection to be mapped to an collection of Futures to be sequenced into a Future of a collection
   * @param fn  the function to apply to the collection to produce an collection of Futures
```

philipschwarz committed:
Update Future.scala

FWIW, I had a go at improving the original proposed description of the 'in' and 'fn' parameters (still too verbose, probably)

philipschwarz commented:
@viktorklang @lrytz FWIW, here are the parameter descriptions as they stand in the latest commit, but with 'IterableOnce' replaced with 'collection' in lines 1, 3, 4 and 5

```Scala
* @tparam A        the type of the value inside the Futures in the collection
* @tparam B        the type of the value of the returned `Future`
* @tparam M        the type of the collection of Futures
* @param in        the collection to be mapped over with the provided function to produce a collection of Futures that is then sequenced into a Future collection
* @param fn        the function to be mapped over the collection to produce a collection of Futures
```

viktorklang commented:
@philipschwarz It's an improvement in my book. @lrytz wdyt?

lrytz commented:
Yes, let's go with that. @philipschwarz can you update the PR?

philipschwarz committed:
Update Future.scala

replaced 'IterableOnce' with 'collection'

philipschwarz commented:
@viktorklang @lrytz done

lrytz commented:
LGTM, can you squash the changes into a single commit?

philipschwarz commented:
@lrytz I created another version of this PR, one that has only one commit: https://github.com/scala/scala/pull/7554 - closing this one
